x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/analyses/Classification.java

```diff
@@ -251,12 +251,13 @@ public boolean supportsMissingValues() {
 
     @Override
     public boolean persistsState() {
-        return false;
+        return true;
     }
 
     @Override
     public String getStateDocId(String jobId) {
-        throw new UnsupportedOperationException();
+        // The state doc id prefix is same as for regression
+        return jobId + "_regression_state#1";
```

Comment:
Shouldn't it be "_classification_state" instead?

Reply:
I wasn't sure whether the underlying state was similar so I ignored this. However, after talking to @tveasey about it we decided it'd be more clear to distinguish between them. So, I'll be raising a PR to address this in the c++ side. Thanks for raising it!